drivers/clock_control/clock_fixed.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (C) 2022 Google, LLC
+ *
+ */
+
+#include <zephyr/drivers/clock_control.h>
+
+#define DT_DRV_COMPAT fixed_clock
+
+static int fixed_clk_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
```

Comment:
This needs to turn on the input clock for the fixed clock, if any (as supported by the binding, see https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/clock/fixed-clock.yaml).

Reply:
Oh! TIL. I didn't realize fixed clocks could have input clocks. Will fix.

Reply:
Done. PTAL

Reply:
@henrikbrixandersen do we have any use-case for this? I've implemented it but it seems kinda hard to wrap my mind around the idea of having a fixed-clock that depends on some other clock.

Reply:
I can't see why a fixed clock could _not_ have an input clock?

Reply:
@henrikbrixandersen I think we should look at fixed-clock and fixed-factor clock separately.

For a fixed-factor clock, I completely agree, it should support parent clocks (since it is only a derived version of it).

For a fixed-clock (previous to this discussion) my admittedly Linux-based understanding was that it's an _always-on_ clock with a _fixed rate/frequency_.
Right now this is reflected by it being able to support them via compile time DT macros.

As such I'm struggling with use-cases for a parent clock in a _fixed-clock_ and propose to simplify the fixed-clock driver to be always-on with fixed rate.

Reply:
> As such I'm struggling with use-cases for a parent clock in a _fixed-clock_ and propose to simplify the fixed-clock driver to be always-on with fixed rate.

Okay, I see your point. Let's simplify the fixed-clock for now and leave out support for input clocks there. You'll need to remove the `clocks` property from the devicetree binding, then.

Reply:
Done. PTAL.